application/oauth_server/service.py

```diff
@@ -192,6 +204,31 @@ def get_smart_service(self, unverified_decoded_jwt):
 
         return smart_service
 
+    def _verify_aud(self, aud: str, auth_client_id):
+        """
+        The client credential token can either be a HTI token or a client auth token
+        :param aud: the value from the auth field
+        :param auth_client_id: the client id set as the issuer of the token
+        :return: True if the aud is valid
+        """
+        token_endpoint = self._get_token_endpoint()
+        if aud.startswith("Device/"): # Check if the audience is the Device/123 reference.
+            smart_service: SmartService = SmartService.query.filter_by(fhir_store_device_id=aud).first()
+            if smart_service and smart_service.client_id == auth_client_id:
+                return True
+
+            logger.warning(
+                f'Unable to find a smart service {aud} with client id {auth_client_id}')
+            return False
+        elif token_endpoint == aud:  # Check if the audience is the auth service itself.
+            return True
+
+        logger.warning(f'Unexpected aud in token: {aud}')
+        return False
+
+    def _get_token_endpoint(self):
+        return request.root_url + 'token'
+
```

Comment:
@rolandgroen Misschien is het goed om in deze functie onderscheid te maken tussen HTI token en client auth token. Nu kan je in principe altijd een HTI launch uitvoeren wanneer je het token endpoint zet als `aud`